### PR TITLE
BAH-1104|BAH-790|Buvaneswari|FormPrivilege table modification

### PR DESCRIPTION
**Pull request by buvaneswari-arun**

### CODE REVIEW

File: api/src/main/java/org/bahmni/module/bahmni/ie/apps/service/impl/BahmniFormPrivilegesServiceImpl.java [hunk 1]
```diff
@@ -59,54 +55,49 @@ public FormPrivilege saveFormPrivilege(FormPrivilege toPersistFormPrivilege) thr
...
-                    }else if (tempEntrySet.getKey().toString().equalsIgnoreCase("formVersion")) {
-                        tempPrivilege.setFormVersion((String) temp.get("formVersion"));
-                        formVersion = tempPrivilege.getFormVersion();
-                    }
+        Iterator privilegeItr = formPrivileges.iterator();
+        int formId = 0;
+        String formVersion = "";
+
+        while (privilegeItr.hasNext()) {
+            LinkedHashMap temp = (LinkedHashMap) privilegeItr.next();
```
Comment (angshu) [inline api/src/main/java/org/bahmni/module/bahmni/ie/apps/service/impl/BahmniFormPrivilegesServiceImpl.java]: Why are we using these temporary variables and looping through the entry?
If the purpose is to just prepare list of "FormPrivilege", just create an instance and copy the values 
1) Optional.ofNullable(temp.get("formId")).ifPresent(x -> { tempPrivilege.setFormId((Integer) x); });
or more conventional
2)  if (temp.get("formId") != null) {
         tempPrivilege.setFormId((Integer) temp.get("formId"));
     }

You don't need variables "formId" and "formVersion" as well. You can just get them "privilegesList.get(0).getFormId()" and "privilegesList.get(0).getFormVersion()" methods. Of course, you ought to make sure that "deleteAllThePrivilegesFromDB()" method checks that the variables are not null (and exit if so)

File: api/src/main/java/org/bahmni/module/bahmni/ie/apps/model/FormPrivilege.java [hunk 2]
```diff
@@ -8,14 +8,45 @@
 import javax.persistence.Entity;
 import javax.persistence.Table;
 import java.io.Serializable;
+import java.util.Date;
 import java.util.List;
 
-public class FormPrivilege extends BaseOpenmrsMetadata {
+public class FormPrivilege extends BaseOpenmrsObject implements java.io.Serializable {
```
Comment (angshu) [inline api/src/main/java/org/bahmni/module/bahmni/ie/apps/model/FormPrivilege.java:11]: Why not derive from "BaseOpenmrsData" 
like we are doing in other places? https://github.com/Bahmni/openmrs-module-appointments/blob/master/api/src/main/java/org/openmrs/module/appointments/model/Speciality.java

Comment (angshu) [reply api/src/main/java/org/bahmni/module/bahmni/ie/apps/model/FormPrivilege.java:11]: We are also not having any traceability of who created the entry, if you do so.

Comment (buvaneswari-arun) [reply api/src/main/java/org/bahmni/module/bahmni/ie/apps/model/FormPrivilege.java:11]: I agree to this point. Will take up this as part of second cut . This needs through investigation as our bahmni.ie.apps hibernate framework has to be changed to accommodate as I am facing issues with lazy initialization, session and bean creation issues when I try to use BaseOpenmrsData and creator.

File: api/src/main/java/org/bahmni/module/bahmni/ie/apps/service/impl/BahmniFormPrivilegesServiceImpl.java (hunk 1 above)
Comment (angshu) [reply api/src/main/java/org/bahmni/module/bahmni/ie/apps/service/impl/BahmniFormPrivilegesServiceImpl.java]: @buvaneswari-arun are you going to work on this part?

Comment (buvaneswari-arun) [reply api/src/main/java/org/bahmni/module/bahmni/ie/apps/service/impl/BahmniFormPrivilegesServiceImpl.java]: This is done.  The changes are committed.

**Review by angshu — APPROVED** [review]

**Pull request merged**